Render "prehistory" document with asciidoc

aspiwack opened the pull request:
## What
The "prehistory" `README` file in `dev/doc/` which was uploaded by @herbelin recently is too important, in my opinion, to be left to rot as a plain text file. This pull request turns this document into an asciidoc file for easier consumption.

## Why asciidoc
It is supported by Github, therefore, there is an immediate link: look at [how pretty it is](https://github.com/aspiwack/coq/blob/render-prehistory/dev/doc/README-V1-V5.asciidoc)!

## Why not markdown
Asciidoc is just as human readable as markdown. In fact it resembles markdown quite a bit. But asciidoc is a full-fledged document format with more formatting options (though some are, unfortunately, not supported by github) and goodies such as table of content and bibliography sections (both of which are used in this PR). While markdown is good for small pieces of text, such as the description of a pull request, asciidoc scales to larger document such as a longish `README` file on the prehistory of a great system, or books published by O'Reilly editions.

## Details
I split my modifications in several commits for easier review. The main commits are

- Formatting: formatting tagging for asciidoc, as well as some typographic improvements (missing diacritics, greek letter written with unicode rather than their latin names, title in italic rather than double-quotes, …)
- Improvement of English: sometimes I felt the English in the document was rather clumsy, so I proposed a few small changes which I believe improve the readability of the text. I'm not entirely sure about the status of this `README` and whether is should be preserved as close as possible to the original documents which served as its sources. I've focussed on making the content as accessible as possible, but opinions may differ.
- Justification: for easier review, I moved justification of plain text in a separate commit. Asciidoc is human readable, so I tried to have the plain text as readable as possible too.
- Bibliography: I moved the detailed bibliographic references into a separate section to make the text lighter and more pleasant to read. (I also added two or three missing references).

## Questions
Some open questions:

- The document features a number of names in capital letters (like CONSTR, INRIA, CAML, PROLOG). I find this unpleasant, and a rather bizarre convention, I'd be in favour of changing them to regular capitalisation (Constr, Inria, Caml, Prolog), though since Inria and Caml are acronyms, it may be considered to leave them in capitals (small capitals, although more appropriate, are not available as a formatting option, as far as I'm aware).
- The first sentence of the document is "This archive contains the sources of the CONSTR ancestor of the Coq proof assistant". Is that really intended?
- Is there a source on the _lambo_ function? The name does not seem very commonly used these days and my biblographic search didn't unearth any digital document which I could use. It would be nice to give some context to the name.
- _Inductive definitions in the Calculus of Constructions_ is listed as a publication in this document but as a user guide on Google scholar. There is a list of user guides just below the list of publications where it is now, should it be moved there?

herbelin commented:
I think this is a good idea. About your questions:
- The intention from Gérard and his co-authors should be respected, such as using capitals letters.
- I updated the initial sentence to clarify the context in which this file has been written. This will also allow us to freely re-format the file as you are proposing to do..
- About "Inductive definitions in the Calculus of Constructions", I don't have a copy of it and I don't know.
- Here is the header of the file lambo.ml from the version 4.2 archive:
      Si f:nat->nat  et n:nat alors lambo(f)(n)=inf{m:nat|f(m)>n}   
      f est supposee croissante et non bornee,	
      l'algorithme developpe est :	
            lambo(f)(n) = if f(0) > n then 0 		
     			 	        else limbo(1)+1	
            where rec limbo(p) = if f(p) > n then 0 	
    					       else 	
    				let inter = limbo(2*p) in 	
    				if f(inter+p) > n then inter		
    						  else inter + p	

To know more, Christine would be the right source.

aspiwack commented:
> The intention from Gérard and his co-authors should be respected, such as using capitals letters.

You may want to check e82fc7cb4104d28619448bde374afde7e32f3dc2 too, I've changed things. I don't believe I've infrigend on the authors' intention, but a second opinion is welcome. In particular [the paragraph on _lambo_](https://github.com/aspiwack/coq/commit/e82fc7cb4104d28619448bde374afde7e32f3dc2#diff-935d0b2fcf181e3250fc73a1679ab35cL50).

> I updated the initial sentence to clarify the context in which this file has been written.

Waiting for your commit/push ;-)

> Here is the header of the file lambo.ml from the version 4.2 archive

Dear lord… is it really a pun on [limbo dancing](https://en.wikipedia.org/wiki/Limbo_(dance))? Cheeky mathematics are a thing.

While looking around I could find conference proceedings where the _lambo_ function is mentioned during question sessions, as something that was believed to be impossible to define in systems at the time. So it was some sort of a preoccupation, and defining it in proto-Coq an important achievement. What I didn't find was the origin of this discussion, maybe it was something that was never written… Google scholar finds very few articles with the word _lambo_ in it. But since those article are not digitally available and the proceedings I was mentionning are paywalled, I couldn't investigate further. I'll probably just ask Christine about it, then.

herbelin commented:
I think this is OK. If you have the opportunity to meet Gérard, he will certainly be pleased to talk with you about this prehistory of Coq (as well as of the present of Coq actually!).

My commit was in 8.5 but a merge finally put in trunk.

herbelin commented:
Merged.
